File: java_solution/LongestCommonPrefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuffer sb = new StringBuffer();
        
        if (strs == null || strs.length == 0) {
            return sb.toString();
        }
        
        String firstWord = strs[0];
        
        for (int i = 0; i < firstWord.length(); i++) {
            for (int j = 1; j < strs.length; j++) {
                if (i >= strs[j].length() || strs[j].charAt(i) != firstWord.charAt(i)) {
                    return sb.toString();
                }
            }
            sb.append(firstWord.charAt(i));
        }
        
        return sb.toString();
    }
}

File: python_solution/LongestCommonPrefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        result = ""
        
        if (strs is None or len(strs) == 0):
            return result
        
        firstWord = strs[0]
        
        for i in range(0, len(firstWord)):
            for j in range(1, len(strs)):
                if (i >= len(strs[j]) or strs[j][i] != firstWord[i]):
                    return result
                
            result = result + firstWord[i]
            
        return result